demos/common/cpp/models/include/models/image_processing_model.h

```diff
@@ -0,0 +1,37 @@
+/*
+// Copyright (C) 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writingb  software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#pragma once
+#include "image_model.h"
+
+class ImageProcessingModel : public ImageModel {
+public:
+    /// Constructor
+    /// @param modelFileName name of model to load
+    /// @param viewInfo size of view of the result of model
+    /// @param useAutoResize - if true, image will be resized by IE.
+    /// Otherwise, image will be preprocessed and resized using OpenCV routines.
+    ImageProcessingModel(const std::string& modelFileName, bool useAutoResize);
+
+    cv::Size getViewSize();
```

Comment:
This isn't about model. It is never used in `ImageProcessingModel` or its children. This should be in main. For models there are already `outHeight` and `outWidth`. Get `viewResult` (if no FLAGS_vheight or FLAGS_vwidth set) value from each `ImageResult`.
`outHeight`, `outWidth`, `outChannels` aren't really needed to be members. These values can be extracted from blob in `postprocess()`. I really don't like an idea of introduction new class just to store 3 values and luckkily there is a way to remove these values.

Reply:
`ImageProcessingModel` deleted. Get view functionality moved to main.cpp.

Reply:
getViewSize() isn't needed. `viewResult` takes results now either form flags or `ImageResult`. As for opening `VideoWriter`, do that when you have first processed image.
There is https://github.com/openvinotoolkit/open_model_zoo/pull/2315. Align with it on `output_resolution` flag. But there are still comments, so I don't think you should do anything about it until it is merged.